### PR TITLE
Issue #2483, #3083 Update Xpaas image streams and templates for OpenShift 3.11

### LINKED ISSUE #2483: Update Xpaas image streams and templates for OpenShift 3.11
We need to check if we need to update any image streams and templates for OpenShift v3.10.0

### PR DESCRIPTION
**Pull request by agajdosi**
Fixes #2483, #3083

- updates 3.9 templates from https://github.com/openshift/openshift-ansible/tree/release-3.9/roles/openshift_examples/files/examples/v3.9
- updates 3.10 templates from https://github.com/openshift/openshift-ansible/tree/release-3.10/roles/openshift_examples/files/examples/v3.10
- adds 3.11 templates from https://github.com/openshift/openshift-ansible/tree/release-3.11/roles/openshift_examples/files/examples/latest
- makes 3.11 the default version to be deployed

Files ignored:
- eap-cd streams and templates
- eap-62 streams and templates

### CODE REVIEW

**Comment by agajdosi** [comment]
retest this please

**Comment by agajdosi** [comment]
retest this please

**Comment by agajdosi** [comment]
retest this please

**Comment by agajdosi** [comment]
retest this please

**Comment by anjannath** [comment]
retest this please

File: test/integration/features/addons/addon-xpaas.feature [hunk 1]
```diff
@@ -29,9 +29,9 @@ which are then available in OpenShift to the user.
 
   Examples: Required information to test the templates
     | project-name  | template-name           | service-name   | http-endpoint | expected-hello                        |
-    | datagrid65    | datagrid65-basic        | datagrid-app   | /             | Welcome to the JBoss Data Grid Server |
-    | eap70         | eap70-basic-s2i         | eap-app        | /index.jsf    | Welcome to JBoss!                     |
-    | eap71         | eap71-tx-recovery-s2i   | eap-app        | /             | Welcome to JBoss EAP 7                |
+    | eap71-basic   | eap71-basic-s2i         | eap-app        | /             | Welcome to JBoss EAP 7                |
+    | eap71-tx      | eap71-tx-recovery-s2i   | eap-app        | /             | Welcome to JBoss EAP 7                |
+    | datagrid72    | datagrid72-basic        | datagrid-app   | /             | Welcome to the JBoss Data Grid Server |
```
[the remark this answers is not in the document]
Comment (agajdosi) [reply test/integration/features/addons/addon-xpaas.feature]: @anjannath These are present in the addon, however newly these streams points to registry.redhat.io into which we do not have login. This needs https://github.com/minishift/minishift/pull/3016 before. I am adding WIP again.

**Comment by anjannath** [comment]
retest this please

**Comment by agajdosi** [comment]
@anjannath This needs the registry.redhat.io addon to be in place, otherwise it cant be merged and tests will fail.

**Comment by agajdosi** [comment]
retest this please

File: centos_ci.sh [hunk 2]
```diff
@@ -345,7 +345,8 @@ function perform_release() {
 }
 
 function perform_pr() {
-  make prerelease synopsis_docs link_check_docs
+  #make prerelease synopsis_docs link_check_docs
+  env
   MINISHIFT_VM_DRIVER=kvm make integration_pr
```
Comment (anjannath) [inline centos_ci.sh]: @agajdosi Could you also apply the following patch to this file.. and use these env variables in the feature file
```diff
 centos_ci.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/centos_ci.sh b/centos_ci.sh
index 3e707a912..8652539a5 100644
--- a/centos_ci.sh
+++ b/centos_ci.sh
@@ -29,7 +29,7 @@ LATEST="latest"
 function load_jenkins_vars() {
   if [ -e "jenkins-env" ]; then
     cat jenkins-env \
-      | grep -E "(JENKINS_URL|GIT_BRANCH|GIT_COMMIT|BUILD_NUMBER|ghprbSourceBranch|ghprbActualCommit|BUILD_URL|ghprbPullId|GH_TOKEN|CICO_API_KEY|API_TOKEN|JOB_NAME|RELEASE_VERSION|GITHUB_TOKEN|REPO|BRANCH|SKIP_INTEGRATION_TEST)=" \
+      | grep -E "(JENKINS_URL|GIT_BRANCH|GIT_COMMIT|BUILD_NUMBER|ghprbSourceBranch|ghprbActualCommit|BUILD_URL|ghprbPullId|GH_TOKEN|CICO_API_KEY|API_TOKEN|JOB_NAME|RELEASE_VERSION|GITHUB_TOKEN|REPO|BRANCH|SKIP_INTEGRATION_TEST|RH_REGISTRY_USERNAME|RH_REGISTRY_PASSWORD)=" \
       | sed 's/^/export /g' \
       > ~/.jenkins-env
     source ~/.jenkins-env
```

Comment (agajdosi) [reply centos_ci.sh]: Hi @anjannath, sure, will do that. Thank you for figuring this out :1st_place_medal:

**Comment by agajdosi** [comment]
retest this please

**Comment by agajdosi** [comment]
Xpaas addon and the addon-xpaas.feature passed in CentOS-CI: https://ci.centos.org/job/minishift-pr/5571/consoleFull.

Going to revert the Makefile back to normal and remove WIP.

**Comment by agajdosi** [comment]
retest this please

**Comment by agajdosi** [comment]
Xpaas addon and the addon-xpaas.feature passed in CentOS-CI: https://ci.centos.org/job/minishift-pr/5571/consoleFull. Removing the WIP now, this PR is ready for review and merge.

**Comment by LalatenduMohanty** [comment]
@agajdosi I do not see change required for #3083. Not sure if I missed it it the PR, let me know.

File: addons/xpaas/xpaas.addon [hunk 3]
```diff
@@ -1,8 +1,11 @@
 # Name: xpaas
 # Description: Imports xPaaS templates
-# Url: https://github.com/openshift/openshift-ansible/tree/release-3.10/roles/openshift_examples/files/examples/v3.10
+# Url: https://github.com/openshift/openshift-ansible/tree/release-3.11/roles/openshift_examples/files/examples/latest
 # Required-Vars: OPENSHIFT_VERSION
-# Var-Defaults: OPENSHIFT_VERSION=3.10
+# Var-Defaults: OPENSHIFT_VERSION=3.11
+# Depends-On: redhat-registry-login
```
Comment (agajdosi) [inline addons/xpaas/xpaas.addon]: Hi @LalatenduMohanty, the dependency is added in here,

**Pull request merged**